### PR TITLE
common generate scripts to the repo

### PR DESCRIPTION
**Pull request by margani007**

### CODE REVIEW

**Comment by PhillypHenning** [comment]
I personally think the `deploy.sh` script should be included. 

We'd need to update the `deploy.sh` to execute the commons scripts and then call `deploy-unique.sh` at the end, in which each subsequent github action we create would have a `deploy-unique.sh` named script that would then execute any additional specialized logic followed by the startup command (which will change between GHAs)

**Comment by mickmcgrath13** [comment]
the tricky part, i think, will be the passing of user-provided environment variables to the docker run scripts.

what about instead just having the deploy script as-is, but instead call an entrypoint generator script in the commons?

that way, we maintain flexibility for the user and the docker run script env vars and also allow for minimal code duplication.

something like:
```
bash /path/to/commons/run_generators.sh
docker run \
-e MY_CUSTOM_VAR=foo \
...
```

then you could run your custom generator stuff like:
```
bash /path/to/commons/run_generators.sh
bash /path/to/action_code/operations/_scripts/generators/run_generators.sh
docker run \
-e MY_CUSTOM_VAR=foo \
...
```

File: operations/_scripts/generate/shorten_identifier.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    # echo "BASE_REMATCH[0]"
+    # echo "${BASH_REMATCH[0]}"
+    # echo "BASE_REMATCH[1]"
+    # echo "${BASH_REMATCH[1]}"
+    # echo "BASE_REMATCH[2]"
+    # echo "${BASH_REMATCH[2]}"
+
+    # echo "==="
+    # echo "IDENTIFIER"
+    # echo "$IDENTIFIER"
```
Comment (armab) [inline operations/_scripts/generate/shorten_identifier.sh:37]: Please remove uneeded/unused debug code.

For someone looking at the code, this makes them think like the code is unfinished.

**Review by armab — CHANGES_REQUESTED** [review]
Thanks a lot for the Pull Request @margani007! 👍 

A few minor changes to address before merging it, besides that good to have these shared code snippets in the commons.

**Comment by PhillypHenning** [comment]
Some thoughts on next steps; 
- Locally test this and a separate GHA
- Add an action.yaml file (Become a composite action)
    - Test the composite in a composite (GHA-deploy-docker-to-ec2 should pull the commons then run as normal)


Certainly some miles to go but this is an excellent start.

**Pull request merged**